packages/account-abstraction/src/SmartAccountAPI.ts

```diff
@@ -182,21 +184,23 @@ export class SmartAccountAPI extends BaseAccountAPI {
    * - if gas or nonce are missing, read them from the chain (note that we can't fill gaslimit before the wallet is created)
    * @param info
    */
-  async createUnsignedUserOp(info: TransactionDetailsForBatchUserOp): Promise<UserOperation> {
+  async createUnsignedUserOp(info: TransactionDetailsForBatchUserOp, vGasLimit?: BigNumber): Promise<UserOperation> {
     const { callData, callGasLimit } = await this.encodeUserOpCallDataAndGasLimit(info)
     console.log(callData, callGasLimit)
 
     const initCode = await this.getInitCode()
 
-    const verificationGasLimit = BigNumber.from(await this.getVerificationGasLimit())
+    let verificationGasLimit = vGasLimit
```

Comment:
is there a reason to pass custom verificationGasLimit override it from outside? who will pass this, are we expecting devs to pass?